libsolidity/ast/Types.cpp

```diff
@@ -1267,7 +1267,11 @@ shared_ptr<FixedPointType const> RationalNumberType::fixedPointType() const
 	if (v > u256(-1))
 		return shared_ptr<FixedPointType const>();
 
-	unsigned totalBits = max(bytesRequired(v), 1u) * 8;
+	boost::optional<unsigned> requiredBytes = bytesRequired(v, 4096);
```

Comment:
I think the issue here might rather be that `bytesRequired` does not handle negative numbers.